packages/bigcommerce-payments-integration/src/BigCommercePaymentsRatePay/BigCommercePaymentsRatePayPaymentMethod.tsx

```diff
@@ -0,0 +1,210 @@
+
+import {
+    PaymentMethodProps,
+    PaymentMethodResolveId,
+    toResolvableComponent,
+} from '@bigcommerce/checkout/payment-integration-api';
+import { DynamicFormField, DynamicFormFieldType, FormContext } from '@bigcommerce/checkout/ui';
+import { FormField } from '@bigcommerce/checkout-sdk';
+import getBigCommercePaymentsRatePayValidationSchema from '../validation-schemas/getBigCommercePaymentsRatePayValidationSchema';
+import { CustomError } from '@bigcommerce/checkout/payment-integration-api';
+import { SpecificError } from '@bigcommerce/checkout/payment-integration-api';
+import { CountryData, getCountryData } from '@bigcommerce/checkout/payment-integration-api';
+
+const PAYMENT_SOURCE_INFO_CANNOT_BE_VERIFIED = 'PAYMENT_SOURCE_INFO_CANNOT_BE_VERIFIED';
+const PAYMENT_SOURCE_DECLINED_BY_PROCESSOR = 'PAYMENT_SOURCE_DECLINED_BY_PROCESSOR';
+const ITEM_CATEGORY_NOT_SUPPORTED_BY_PAYMENT_SOURCE = 'ITEM_CATEGORY_NOT_SUPPORTED_BY_PAYMENT_SOURCE';
+
+interface RatePayFieldValues {
+    ratepayBirthDate: {
+        getDate(): number;
+        getMonth(): number;
+        getFullYear(): number;
+    };
+    ratepayPhoneCountryCode: string;
+    ratepayPhoneNumber: string;
+}
+
+const formFieldData: FormField[] = [
+    {
+        name: 'ratepayBirthDate',
+        custom: false,
+        id: 'ratepayBirthDate',
+        label: 'payment.ratepay.birth_date',
+        required: true,
+        fieldType: DynamicFormFieldType.DATE,
+        inputDateFormat: 'dd.MM.yyyy',
+    },
+    {
+        name: 'ratepayPhoneCountryCode',
+        custom: false,
+        id: 'ratepayPhoneCountryCode',
+        label: 'payment.ratepay.phone_country_code',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        type: 'string',
+    },
+    {
+        name: 'ratepayPhoneNumber',
+        custom: false,
+        id: 'ratepayPhoneNumber',
+        label: 'payment.ratepay.phone_number',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+    }
+];
+
+const BigCommercePaymentsRatePayPaymentMethod: FunctionComponent<any> = ({
+    method,
+    checkoutService,
+    onUnhandledError,
+    paymentForm: {
+        isSubmitted,
+        setFieldValue,
+        setValidationSchema,
+        setSubmitted,
+    },
+    language,
+    checkoutState,
+}) => {
+    const fieldsValues = useRef<Partial<RatePayFieldValues>>({});
+    const isPaymentDataRequired = checkoutState.data.isPaymentDataRequired();
+    const getCountryInfo = (): CountryData => {
+        const billing = checkoutState.data.getBillingAddress();
+
+        return getCountryData(billing.country)[0] || '';
+    };
+
+    if (!isPaymentDataRequired) {
+        return null;
+    }
+
+    const initializePayment = async () => {
+        try {
+            await checkoutService.initializePayment({
+                gatewayId: method.gateway,
+                methodId: method.id,
+                bigcommerce_payments_ratepay: {
```

Comment:
This is something that should be synchronised with checkout-sdk implementation (better to check with dev who is working with sdk)